src/web/wonka_observable.re

```diff
@@ -0,0 +1,110 @@
+open Wonka_types;
+open Wonka_helpers;
+
+let observableSymbol: string = [%raw
+  {|
+  typeof Symbol === 'function' && Symbol.observable || '@@observable'
+|}
+];
+
+type subscriptionT = {. [@bs.meth] "unsubscribe": unit => unit};
```

Comment:
So the only real alternative to using `Js.t` signatures for this would be to use `[@bs.deriving abstract]`. In that case, something like `type subscriptionT` would look like this:

```reason
[@bs.deriving abstract]
type subscriptionT = {
   unsubscribe: unit => unit
};
```

Then, to use it in something like `fromObservable` below, you'd do the following:

```reason
sink(.
      Start(
        (. signal) =>
          switch (signal) {
          | Close => subscription->unsubscribeGet()
          | _ => ()
          },
      ),
```

I honestly don't _love_ the `[@bs.deriving abstract]` syntax vs. the `Js.t` syntax, so take it or leave it. There was a push a while back to deprecate `Js.t`, but the community likes it too much 😂

Reply:
I started out with that but it doesn't guarantee that the method will be called directly so it might cause issues with Observables that have unbound methods accessing "this"